### PR TITLE
Benchmarks: Add Benchmark - Add gemm flops microbenchmark for amd

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
Add gemm flops microbenchmark for amd.

**Major Revision**
- Add gemm flops microbenchmark for amd.
- Add related example and test file.

### CODE REVIEW

File: superbench/benchmarks/micro_benchmarks/rocm_gemm_flops_performance.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        if not super()._preprocess():
+            return False
+
+        transpose_types = ['N', 'T', 'C']
+        if self._args.transposeA not in transpose_types or self._args.transposeB not in transpose_types:
+            self._result.set_return_code(ReturnCode.INVALID_ARGUMENT)
+            logger.warning(
+                'Unsupported transpose type - benchmark: {}, expected: {}.'.format(self._name, transpose_types)
+            )
+            return False
```
Comment (guoshzhao) [inline superbench/benchmarks/micro_benchmarks/rocm_gemm_flops_performance.py:108]: using choices=['N', 'T', 'C'] in argparse?

Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/rocm_gemm_flops_performance.py]: added, thanks for your suggestion

File: superbench/benchmarks/micro_benchmarks/rocm_gemm_flops_performance.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        super().__init__(name, parameters)
+
+        self._bin_name = 'rocblas-bench'
+        self._support_precisions = ['FP64', 'TF32_TC', 'FP16_TC', 'BF16_TC', 'INT8_TC']
+        self.__precision_and_kernel_map = {
+            'FP64': '-r f64_r -f gemm',
+            'TF32_TC': '-r f32_r -f gemm_ex --compute_type f32_r',
+            'FP16_TC': '-r f16_r -f gemm_ex --compute_type f32_r',
+            'BF16_TC': '-r bf16_r -f gemm_ex --compute_type f32_r',
+            'INT8_TC': '--a_type i8_r --b_type i8_r --c_type i32_r --d_type i32_r -f gemm_ex --compute_type i32_r'
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/rocm_gemm_flops_performance.py:31]: TC is incorrect because it's nvidia term. We should change it to xDLOPS

Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/rocm_gemm_flops_performance.py]: renamed, thanks

**Review by cp5555 — APPROVED** [review]

**Review by guoshzhao — APPROVED** [review]

**Pull request merged**